src/main/java/org/qubership/itool/di/MergerModule.java

```diff
@@ -79,33 +93,41 @@ public List<GraphProcessorTask> provideNormalizationTasksList() {
      * Provides a list of finalization tasks in the order they should be executed.
      * The order is important as some tasks may depend on the results of previous tasks.
      *
-     * @param graphReportProvider Provider for GraphReport
+     * @param createTransitiveQueueDependenciesTaskProvider Provider for CreateTransitiveQueueDependenciesTask
+     * @param createTransitiveHttpDependenciesTaskProvider Provider for CreateTransitiveHttpDependenciesTask
+     * @param recreateDomainsStructureTaskProvider Provider for RecreateDomainsStructureTask
+     * @param recreateHttpDependenciesTaskProvider Provider for RecreateHttpDependenciesTask
      * @return A list of finalization tasks in execution order
      */
     @Provides
     @FinalizationTasks
     public List<GraphProcessorTask> provideFinalizationTasks(
-            Provider<GraphReport> graphReportProvider) {
+            Provider<CreateTransitiveQueueDependenciesTask> createTransitiveQueueDependenciesTaskProvider,
+            Provider<CreateTransitiveHttpDependenciesTask> createTransitiveHttpDependenciesTaskProvider,
+            Provider<RecreateDomainsStructureTask> recreateDomainsStructureTaskProvider,
+            Provider<RecreateHttpDependenciesTask> recreateHttpDependenciesTaskProvider) {
         return List.of(
-            new RecreateHttpDependenciesTask(),
-            new CreateTransitiveQueueDependenciesTask(),
-            new CreateTransitiveHttpDependenciesTask(),
-            new RecreateDomainsStructureTask(graphReportProvider)
+            recreateHttpDependenciesTaskProvider.get(),
+            createTransitiveQueueDependenciesTaskProvider.get(),
+            createTransitiveHttpDependenciesTaskProvider.get(),
+            recreateDomainsStructureTaskProvider.get()
```

Comment:
Have you considered creating a marker interface or marker annotation for the tasks, so [multibindings](https://github.com/google/guice/wiki/Multibindings) could be used instead of manually listing the providers?

Currently, there's no way to add an extra "finalization task".